bokehjs/src/coffee/models/tools/gestures/tap_tool.coffee

```diff
@@ -35,14 +33,13 @@ class TapToolView extends SelectTool.View
       else
         did_hit = sm.inspect(@, view, geometry, {geometry: geometry})
 
-      if did_hit and callback?
-        if _.isFunction(callback)
-          callback(ds, cb_data)
-        else
-          callback.execute(ds, cb_data)
```

Comment:
passing the DataSource to the `execute` method (instead of the model instance that issues the callback), is inconsistent with other Bokeh callbacks.

It's required for the `OpenURL` callback, but also means this is the only callback where `OpenURL` will actually work AFAICT. We need to make these function signatures consistent (between `CustomJS` and `OpenURL`) so that either can be used across Bokeh.